editor/editor_resource_preview.cpp

```diff
@@ -221,7 +221,9 @@ void EditorResourcePreview::_generate_preview(Ref<ImageTexture> &r_texture, Ref<
 			r_small_texture->set_image(small_image);
 		}
 
-		break;
+		if (generated.is_valid()) {
```

Comment:
Note that this only checks for a non-null texture, it doesn't check for a texture of 0x0 size. So still docs vs behavior might be a little misleading.